clients/js/src/embeddings/OpenAIEmbeddingFunction.ts

```diff
@@ -1,45 +1,123 @@
-import { IEmbeddingFunction } from "./IEmbeddingFunction";
+import {IEmbeddingFunction} from "./IEmbeddingFunction";
+
 let OpenAIApi: any;
+let openAiVersion = null;
+let openAiMajorVersion = null;
+
+// interface EmbeddingResponse {
+//     object: string;
+//     data: { embedding: number[] }[];
+//     model: string;
+//     usage: { prompt_tokens: number; total_tokens: number };
+// }
+
+interface OpenAIAPI {
+    createEmbedding: (params: {
+        model: string;
+        input: string[];
+        user?: string;
+    }) => Promise<object>;
+}
+
+class OpenAIAPIv3 implements OpenAIAPI {
+    private readonly configuration: any;
+    private openai: any;
+
+    constructor(configuration: { organization: string, apiKey: string }) {
+        this.configuration = new OpenAIApi.Configuration({
+            organization: configuration.organization,
+            apiKey: configuration.apiKey,
+        });
+        this.openai = new OpenAIApi.OpenAIApi(this.configuration);
+    }
+
+    public async createEmbedding(params: {
+        model: string,
+        input: string[],
+        user?: string
+    }): Promise<object> {
+        const response = await this.openai.createEmbedding({
+            model: params.model,
+            input: params.input,
+        });
+        return response.data;
+    }
+}
+
+class OpenAIAPIv4 implements OpenAIAPI {
+    private readonly apiKey: any;
+    private openai: any;
+
+    constructor(apiKey: any) {
+        this.apiKey = apiKey;
+        this.openai = new OpenAIApi({
+            apiKey: this.apiKey,
+        });
+    }
+
+    public async createEmbedding(params: {
+        model: string,
+        input: string[],
+        user?: string
+    }): Promise<object> {
+        return await this.openai.embeddings.create(params);
+    }
+}
 
 export class OpenAIEmbeddingFunction implements IEmbeddingFunction {
     private api_key: string;
     private org_id: string;
     private model: string;
+    private openaiApi: OpenAIAPI;
 
     constructor({
-        openai_api_key,
-        openai_model,
-        openai_organization_id,
-    }: {
+                    openai_api_key,
+                    openai_model,
+                    openai_organization_id,
+                }: {
         openai_api_key: string,
         openai_model?: string,
         openai_organization_id?: string
     }) {
         try {
             // eslint-disable-next-line global-require,import/no-extraneous-dependencies
             OpenAIApi = require("openai");
-        } catch {
-            throw new Error(
-                "Please install the openai package to use the OpenAIEmbeddingFunction, `npm install -S openai`"
-            );
+            const fs = require('fs');
```

Comment:
is there a simpler way to get the version directly from the import?

eg for 0.4.0+ -- i think we may be able to access this - https://github.com/openai/openai-node/blob/master/src/version.ts

Reply:
It could work for @4 but 3 doesn't have it:

<img width="1197" alt="image" src="https://github.com/chroma-core/chroma/assets/1157440/4ece23f8-5caa-49a0-961a-5df1182b46d9">


I can try to access and if exception is raised then I'll know :)